docs/cookbook/agents/web_search_agent.md

```diff
@@ -24,107 +24,240 @@ Make sure to also set your `OPENAI_API_KEY` if you haven't already. We are using
 
 ## Add DuckDuckGo Tool
 
-The first step is to create a `WebAssistant` that first conducts a web search based on the user's query. Let’s go ahead and add our web search tool:
+The first step is to create a `WebAssistant` that first conducts a web search based on the user's query. Let’s go ahead and create our `WebAssistant` and add our web search tool:
 
 ```python
 import requests
-from bs4 import BeautifulSoup
 from duckduckgo_search import DDGS
+from openai.types.chat import ChatCompletionMessageParam
+from pydantic import BaseModel
+
+class WebAssistant(BaseModel):
+    messages: list[ChatCompletionMessageParam] = []
+    search_history: list[str] = []
+
+    def _web_search(self, queries: list[str]) -> str:
+        """Search the web for the given text and parse the paragraphs of the results.
+
+        Args:
+            queries: The list of queries to search for.
+
+        Returns:
+            Parsed paragraphs of each of the webpages, separated by newlines.
```

Comment:
this description seems wrong? the return is just the joined `href` links but it seems like this function doesn't include the body and is definitely not parsing anything?

the function body e.g. `parsed_results` should also be updated to be more clear around what the function is doing